core/src/main/java/com/novoda/espresso/TalkBackTestRule.java

```diff
@@ -27,16 +33,32 @@ public Statement apply(Statement base, Description description) {
         @Override
         public void evaluate() throws Throwable {
             talkBackStateSettingRequester.requestEnableTalkBack();
-            sleepToAllowTalkBackServiceToChangeState();
+            sleepUntilTalkBackIsEnabled();
 
             baseStatement.evaluate();
 
             talkBackStateSettingRequester.requestDisableTalkBack();
-            sleepToAllowTalkBackServiceToChangeState();
+            sleepUntilTalkBackIsDisabled();
+        }
+
+        private void sleepUntilTalkBackIsEnabled() {
+            while (talkBackDisabled()) {
```

Comment:
maybe add a timeout? otherwise it could potentially sleep forever

Reply:
i.e. stop after 150 iterations regardless of the state of talkback

Reply:
I'm not sure this is true - I will check by adding a 30min timeout and watching a movie.

Reply:
commit 3d3e7c5e78772c1e9e8192ad0772ae9f58858e87
Author: Ataul Munim <hello@ataulm.com>
Date:   Fri Jan 12 15:14:39 2018 +0000

    Add limit to retries for waiting for TalkBack toggle until test fails


https://github.com/novoda/espresso-support/pull/15#issuecomment-357265279